File: 天翼云/tianyiyun.js
/**
 * comParam_curTime
 * comParam_seqCode
 * comParam_signature
 * password
 */
const CryptoJS = require('crypto-js')

function get_com_param_cur_time() {
    return ((new Date).getTime() + 124).toString()
}

function get_com_param_seq_code() {
    var e, n, t = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : 32,
        r = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : 16,
        a = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz".split(""), o = [];
    if (r = r || a.length,
        t)
        for (e = 0; e < t; e++)
            o[e] = a[0 | Math.random() * r];
    else
        for (o[8] = o[13] = o[18] = o[23] = "-",
                 o[14] = "4",
                 e = 0; e < 36; e++)
            o[e] || (n = 0 | 16 * Math.random(),
                o[e] = a[19 === e ? 3 & n | 8 : n]);
    return o.join("")
}

// console.log(get_com_param_cur_time())
// console.log(get_com_param_seq_code())

function get_com_param_signature(n, r) {
    // n = 1714746766671;
    t = "s54zv9bm1vd5czfujy6nnuxj1l4g2ny6";
    // r = 'B25FED9E01E243CCC304188688352121';
    // console.log(CryptoJS.MD5(n + r + CryptoJS.MD5(r + t + n)).toString())
    return (CryptoJS.MD5(n + r + CryptoJS.MD5(r + t + n)).toString())
}

function _(e) {
    var n = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : {};
    if (e && "string" === typeof e) {
        var t = n.text || "0"
            , r = n.length || 24;
        if (e.length < r)
            for (var a = e.length; a < r; a++)
                e += t;
        else
            e = e.substring(0, r);
        return e
    }
}

function get_password(password, username) {
    // password = '12345asdf'
    username = _(username)

    key = CryptoJS.enc.Utf8.parse(username)

    cfg = {
        mode: CryptoJS.mode.ECB,
        padding: CryptoJS.pad.Pkcs7
    }
    var ciphertext = CryptoJS.TripleDES.encrypt(password, key, cfg);

    return ciphertext.toString()
}

File: 天翼云/tianyiyun.py
import requests
import execjs

with open('./tianyiyun.js', 'r') as fp:
    js_code = fp.read()

js_engine = execjs.compile(js_code)

comParam_curTime = js_engine.call('get_com_param_cur_time')
comParam_seqCode = js_engine.call('get_com_param_seq_code')
comParam_signature = js_engine.call('get_com_param_signature', comParam_curTime, comParam_seqCode)

cookies = {
    'ct_m_dvc_id': '903dacbe8891e05d87544f4af62641e5',
    'sid1': '1714740900248-509FE8CC0EC46414E4DA7628CC0CB805',
    'sid2': '1714740900248-509FE8CC0EC46414E4DA7628CC0CB805',
    'pvid': '1',
}

headers = {
    'Accept': 'application/json, text/plain, */*',
    'Accept-Language': 'zh-CN,zh;q=0.9',
    'Cache-Control': 'no-cache',
    'Connection': 'keep-alive',
    'Content-Type': 'application/x-www-form-urlencoded',
    # 'Cookie': 'ct_m_dvc_id=903dacbe8891e05d87544f4af62641e5; sid1=1714740900248-509FE8CC0EC46414E4DA7628CC0CB805; sid2=1714740900248-509FE8CC0EC46414E4DA7628CC0CB805; pvid=1',
    'Origin': 'https://m.ctyun.cn',
    'Pragma': 'no-cache',
    'Referer': 'https://m.ctyun.cn/wap/main/auth/login?redirect=%2Fmy',
    'Sec-Fetch-Dest': 'empty',
    'Sec-Fetch-Mode': 'cors',
    'Sec-Fetch-Site': 'same-origin',
    'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36',
    'sec-ch-ua': '"Chromium";v="124", "Google Chrome";v="124", "Not-A.Brand";v="99"',
    'sec-ch-ua-mobile': '?0',
    'sec-ch-ua-platform': '"macOS"',
    'x-riskdevicesign': '903dacbe8891e05d87544f4af62641e5',
}

params = {
    'referrer': 'wap',
    'mainVersion': '300031500',
    # 'comParam_curTime': '1714741417080',
    'comParam_curTime': comParam_curTime,
    # 'comParam_seqCode': '2A355DEF1130A845502AD5B61E277792',
    'comParam_seqCode': comParam_seqCode,
    # 'comParam_signature': '94a065b7af6e1b177e44a9c1c75f6e2b',
    'comParam_signature': comParam_signature,
    'isCheck': 'true',
    'locale': 'zh-cn',
}

username = '15512344321@163.com'
password = '12345asdf'
enc_password = js_engine.call('get_password', password, username)

data = {
    'userName': username,
    'password': enc_password,
    'authCode': '',
}

response = requests.post('https://m.ctyun.cn/account/login', params=params, cookies=cookies, headers=headers, data=data)
print(response.text)
